amp/WEB-INF/src/org/digijava/module/aim/util/LuceneUtil.java

```diff
@@ -741,60 +743,34 @@ public Integer execute(Connection conn) throws SQLException {
         return res;
     }
 
-    static void listDocuments(IndexReader indexReader)
-    {
-        try
-        {
-            int nrDocs = indexReader.numDocs();
-            for(int i = 0; i < nrDocs; i++)
-            {
-                Document doc = indexReader.document(i);
-                Map<String, String> document = digestDocument(doc);
-                System.out.format("Document #%d has fields: %s\n", i, document.toString());
-            }
-        }
-        catch(Exception e)
-        {
-            logger.error("error listing Lucene documents", e);
-        }
-    }
-
-    public static int deletePledge(String idx, String field, String search){
-        Term term = new Term(field, search);
-        IndexReader indexReader;
-        try {
-            
-            indexReader = IndexReader.open(FSDirectory.open(new File(idx)), false);
-            //listDocuments(indexReader);
-            int ret = indexReader.deleteDocuments(term);
-            indexReader.close();
-            return ret;
-        } catch (Exception e) {
-            logger.error("error deleting pledge from Lucene index", e);
-            return 0;
-        }
-    }
-
-    public static int deleteEntry(String idx, String field, String search){
+    public static boolean deleteEntry(String idx, String field, String search) {
         Term term = new Term(field, search);
-        IndexReader indexReader;
+        IndexWriter indexWriter = null;
         try {
-            indexReader = IndexReader.open(FSDirectory.open(new File(idx)), false);
-            //listDocuments(indexReader);
-            int ret = indexReader.deleteDocuments(term);
-            indexReader.close();
-            return ret;
+            IndexWriterConfig indexWriterConfig = new IndexWriterConfig(Version.LUCENE_36, LuceneUtil.ANALYZER);
+            Directory directory = FSDirectory.open(new File(idx));
```

Comment:
we need to set LOCK_OBTAIN_WAIT_FOREVER here too. If you post a lot of updates in parallel you get the cannot acquire lock exception. I tried with 5 and I got the exception